azurerm/internal/services/datashare/tests/resource_arm_data_share_dataset_kusto_database_test.go

```diff
@@ -0,0 +1,130 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+)
+
+func TestAccAzureRMDataShareDataSetKustoDatabase_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_data_share_dataset_kusto_database", "test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDataShareDataSetDestroy("azurerm_data_share_dataset_kusto_database"),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDataShareDataSetKustoDatabase_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDataShareDataSetExists(data.ResourceName),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "display_name"),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "kusto_cluster_location"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMDataShareDataSetKustoDatabase_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_data_share_dataset_kusto_database", "test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMDataShareDataSetDestroy("azurerm_data_share_dataset_kusto_database"),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMDataShareDataSetKustoDatabase_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMDataShareDataSetExists(data.ResourceName),
+				),
+			},
+			data.RequiresImportErrorStep(testAccAzureRMDataShareDataSetKustoDatabase_requiresImport),
+		},
+	})
+}
+
+func testAccAzureRMDataShareDataSetKustoDatabase_template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctest-datashare-%[1]d"
```

Comment:
we should start this with
```suggestion
  name     = "acctestRG-datashare-%[1]d"
```